### ISSUE TITLE
Triton provider keeps waiting for failed provision job to complete

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-triton#1]
_This issue was originally opened by @StanYago as hashicorp/terraform#13447. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

### Terraform Version
Terraform v0.9.2

### Affected Resource(s)
- triton_machine

### Terraform Configuration Files
```hcl
resource "triton_machine" "test-kafka" {
  name    = "test-kafka"
  package = "kafka"
  image   = "${var.base64-lts}"
}
```

### Expected Behavior
Triton job failed.  Terraform should have been notified of this failure. 

### Actual Behavior
Terraform kept waiting for job to complete for 10m waiting for triton_machine to finish.

### Steps to Reproduce
1. Create a package with previously undefined trait in Triton
2. Specify the package in terraform
3. `terraform apply`

### Important Factoids
Potentially any failure in provisioner is not reflected back in terraform.

**Comment by StanYago** [comment]
When terraform triggered provisioning job fails in triton (which could happen due to lack of matching resources defined in a package), terraform does not receive the error and waits for 10 minutes before failing.  Triton creates UUID and returns that to terraform which terraform marks as tainted. On the next run terraform will attempt to destroy tainted VM that is already marked as failed in Triton. In this case no job is created in triton, but terraform shows "Still destroying... (ID: ef54a447-...-...-... )" for the next 10 minutes.

**Comment by cheapRoc** [comment]
This was released in v0.1.1.

**Comment by sean-** [comment]
Closing issue now that this has been addressed and shipped in a released version.  Thank you!
